modules/thoughtleadrBidAdapter.js

```diff
@@ -1,29 +1,66 @@
-'use strict';
 var bidfactory = require('src/bidfactory');
 var bidmanager = require('src/bidmanager');
 var utils = require('src/utils');
-var adloader_1 = require('src/adloader');
+var ajax_1 = require('src/ajax');
 var adaptermanager = require('src/adaptermanager');
-var ROOT_URL = '//cdn.thoughtleadr.com/v4/';
+
+var COOKIE_SYNC_ID = 'tldr-cookie-sync-div';
+var UID_KEY = 'tldr_uid';
+var URL_API = 'tldr' in window && tldr.config.root_url ? tldr.config.root_url : '//a.thoughtleadr.com/v4/';
+var URL_CDN = 'tldr' in window && tldr.config.cdn_url ? tldr.config.cdn_url : '//cdn.thoughtleadr.com/v4/';
 var BID_AVAILABLE = 1;
+var BID_UNAVAILABLE = 2;
+
+function getVal(key) {
+  if ('localStorage' in window) {
+    return localStorage[key];
+  }
+  if ('sessionStorage' in window) {
+    return sessionStorage[key];
+  }
+  return null;
+}
+
+function setVal(key, val) {
+  if ('localStorage' in window) {
+    localStorage[key] = val;
+  }
+  if ('sessionStorage' in window) {
+    sessionStorage[key] = val;
+  }
+}
+
+function getUid() {
+  var uid = getVal(UID_KEY);
+  if (!uid) {
+    uid = utils.generateUUID(null);
+    setVal(UID_KEY, uid);
+  }
+  return uid;
+}
+
+function writeFriendlyFrame(html, container) {
+  const iframe = document.createElement("iframe");
+  iframe.style.width = "0";
+  iframe.style.height = "0";
+  iframe.style.border = "0";
+
+  iframe.src = "javascript:false";
+  container.appendChild(iframe);
+
+  const doc = iframe.contentWindow.document;
+  doc.open().write(`<html><head></head><body>${html}</body></html>`);
```

Comment:
Ok, that's fine.

I chatted with our team lead, and while he confirmed that there _is_ a security issue here, we have the same vulnerability throughout prebid.

We're changing this in #891, when we'll merge #1229 and then (at some point in the future) prevent adapters from executing any external JS at all... but for now, we're leaving it to publishers to only load bidders whose code they trust.